app/models/manageiq/providers/amazon/storage_manager/s3/cloud_object_store_container.rb

```diff
@@ -45,11 +72,8 @@ def raw_cloud_object_store_container_clear
 
   def self.raw_cloud_object_store_container_create(ext_management_system, options)
     # frontend stores name as :name, but amazon expects it as :bucket
-    options[:bucket] = options.delete(:name) unless options[:name].nil?
-
-    region = options[:create_bucket_configuration][:location_constraint]
-    connection = ext_management_system.connect(:region => region)
-    bucket = connection.create_bucket(options)
+    connection = ext_management_system.connect(:region => options[:region])
+    bucket = connection.create_bucket(:bucket => options[:name], :create_bucket_configuration => {:location_constraint => options[:region]})
```

Comment:
for some reason I was expecting `with_provider_object`.

Not sure if `with_provider_object` works for us here, but I guess I always thought it would close the connection or something.

Reply:
ooh. I see. you converted `:name` to `:bucket` here.